File: src/transformers/features/json/JsonAssertParseTransformer.ts
import { JsonAssertParseProgrammer } from "../../../programmers/json/JsonAssertParseProgrammer";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonAssertParseTransformer {
    export const transform = GenericTransformer.scalar("assertParse")(
        (project) => (modulo) =>
            JsonAssertParseProgrammer.write(project)(modulo),
    );
}

File: packages/transform/src/features/json/JsonAssertParseTransformer.go
package json

import (
	"github.com/samchon/typia/packages/core/src/programmers"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var JsonAssertParseTransformer = jsonAssertParseTransformerNamespace{}

type jsonAssertParseTransformerNamespace struct{}

func (jsonAssertParseTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return "(input => " + programmers.AssertProgrammer.Write(call.TypeText) + "(JSON.parse(input)))"
}
